xbmc/platform/android/activity/XBMCApp.cpp

```diff
@@ -1621,3 +1624,41 @@ void CXBMCApp::surfaceDestroyed(CJNISurfaceHolder holder)
 
   m_window.reset();
 }
+
+void CXBMCApp::OnVoiceRecognition()
+{
+  if (CJNISpeechRecognizer::isRecognitionAvailable(CXBMCApp::Get()))
+  {
+    if (CJNIContext::checkCallingOrSelfPermission("android.permission.RECORD_AUDIO") == CJNIPackageManager::PERMISSION_GRANTED) 
+    {
+      // must be called from the main thread
```

Comment:
why?

Reply:
This requirement is indicated in the [SpeechRecognizer](https://developer.android.com/reference/android/speech/SpeechRecognizer) documentation

Reply:
okay, could you add a link to the documentation to the comment, please?